bundle/src/main/java/com/cognifide/cq/cqsm/core/history/HistoryEntryWriter.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * ========================LICENSE_START=================================
+ * AEM Permission Management
+ * %%
+ * Copyright (C) 2013 Cognifide Limited
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =========================LICENSE_END==================================
+ */
+
+package com.cognifide.cq.cqsm.core.history;
+
+import com.cognifide.cq.cqsm.api.history.HistoryEntry;
+import java.util.Calendar;
+import lombok.Builder;
+import org.apache.commons.lang.StringUtils;
+import org.apache.sling.api.resource.ModifiableValueMap;
+import org.apache.sling.api.resource.Resource;
+
+@Builder
+public class HistoryEntryWriter {
+
+  private String author;
+  private Calendar executionTime;
+  private String executor;
+  private String fileName;
+  private String filePath;
+  private String instanceType;
+  private String instanceHostname;
+  private Boolean isRunSuccessful;
+  private String mode;
+  private String progressLog;
+
+  public void writeTo(Resource historyLogResource) {
+    ModifiableValueMap valueMap = historyLogResource.adaptTo(ModifiableValueMap.class);

```

Comment:
maybe we should clear the map first?

Reply:
I don't want to remove jcr's default properties. And this class is always saving the same properties, so old values will be overridden.